packages/landing/src/components/layer.switcher.dropdown.tsx

```diff
@@ -228,25 +230,33 @@ export class LayerSwitcherDropdown extends Component<unknown, LayerSwitcherDropd
 }
 
 /**
- * Determine if the bounds in EPSG:3857 intersects the provided layer
- *
- * TODO: It would be good to then use a more comprehensive intersection if the bounding box intersects,
- * there are complex polygons inside the attribution layer that could be used but they do not have all
- * the polygons
+ * Determine if the polygon intersects the provided layer
  *
- * @param bounds Bounding box in EPSG:3857
+ * @param bounds polygon to check
  * @param layer layer to check
  * @returns true if it intersects, false otherwise
  */
-function doesLayerIntersect(bounds: Bounds, layer: LayerInfo): boolean {
+function doesLayerIntersect(bounds: MultiPolygon, layer: LayerInfo): boolean {
   // No layer information assume it intersects
   if (layer.lowerRight == null || layer.upperLeft == null) return true;
 
-  // It is somewhat easier to find intersections in EPSG:3857
-  const ul3857 = Projection.get(GoogleTms).fromWgs84(layer.upperLeft);
-  const lr3857 = Projection.get(GoogleTms).fromWgs84(layer.lowerRight);
+  const poly = Wgs84.bboxToMultiPolygon([
+    layer.lowerRight[0],
+    layer.upperLeft[1],
+    layer.upperLeft[0],
+    layer.lowerRight[1],
+  ]);
+
+  const inter = intersection(bounds, poly);
+  if (inter == null || inter.length === 0) return false;
+
+  // No attribution state loaded, assume it intersects
+  const attrs = MapAttrState.attrsSync.get('all');
+  if (attrs == null) return true;
 
-  const layerBounds = Bounds.fromBbox([ul3857[0], ul3857[1], lr3857[0], lr3857[1]]);
+  const attrLayer = attrs.attributions.filter((f) => f.collection.title === layer.title);
```

Comment:
I am not a huge fan of using the title to find a match between the attribution.json and the WMTSCapabilities.xml

I would prefer we add the `layer.name` to attribution.json and/or the `layer.id` to the WMTS

I am leaning towards layer.name to attribution.json, but as its a public API I would like to do that as a seperate PR